### PR TITLE
LSP: Fixed problem with workspace/symbols called before first JavacCompiler created.

### PR DESCRIPTION
**Pull request by dbalek**
The problem was with `ParserManager.parseWhenScanFinished("text/x-java", new UserTask() {...` called before the first `JavacParser` was created by `java.source` infrastructure. Since `JavacParser` cannot be created without supplying a `ClasspathInfo`, exception was thrown and the given `UserTask` was not executed. Fixed by supplying the `ClasspathInfo` based on the opened projects.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

**Comment by lahodaj** [comment]
I don't mind this patch, but I wonder if there would be opportunity to solve this in the infrastructure, e.g. by synthesizing some ClasspathInfo when none is provided for this task?

**Comment by dbalek** [comment]
> I don't mind this patch, but I wonder if there would be opportunity to solve this in the infrastructure, e.g. by synthesizing some ClasspathInfo when none is provided for this task?

It is hard to create any ClasspathInfo without FileObject (before any project or file gets opened)

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java [hunk 1]
```diff
@@ -970,175 +966,182 @@ public boolean cancel(boolean mayInterruptIfRunning) {
...
         server.openedProjects().thenAccept(openedProjects -> {
-            try {
-                ParserManager.parseWhenScanFinished("text/x-java", new UserTask() {
-                    @Override
-                    public void run(ResultIterator resultIterator) throws Exception {
-                        WORKER.post(() -> {
-                            try {
-                                List<WorkspaceSymbol> symbols = new ArrayList<>();
-                                SearchType searchType = getSearchType(queryFin, exactFin, false, null, null);
+            if (openedProjects.length < 0) {
```
Comment (sdedic) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java:969]: < 1 ?

**Review by sdedic — APPROVED** [review]

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java (hunk 1 above)
Comment (dbalek) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java]: OOPS - fixed.

**Pull request merged**